### PR TITLE
 capture-webcam-video #286

### LINKED ISSUE #286: capture-webcam-video
## Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Checked that your rule idea isn't already filed: [search](https://github.com/fireeye/capa-rules/issues?q=is%3Aissue+is%3Aopen+)

## Summary

Let's have a rule that detects a system's webcam being used to capture video

## Examples

TBD

## Features

I'd need to look around for real world examples that do this, but it seems like we could look for calling `capCreateCaptureWindow` + calling `capCaptureSequence` + (calling `capFileSaveAs` or calling `capFileSetCaptureFile`)

## Additional context

N/A

### Namespace
collection/webcam

### References
- https://docs.microsoft.com/en-us/windows/win32/multimedia/capturing-data
- https://docs.microsoft.com/en-us/windows/win32/multimedia/naming-the-capture-file

### Other rule meta information

N/A

### PR DESCRIPTION
**Pull request by johnk3r**
closes #286 

Hello guys,

unfortunately I couldn't find a sample that met the logic proposed in the ISSUE, for this reason I submitted it as "nursery".

### CODE REVIEW

File: nursery/capture-webcam-video.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            - api: SendMessage
+            - number: 0x43E = WM_CAP_SEQUENCE
+        - basic block:
+          - and:
+            - api: SendMessage
+            - number: 0x417 = WM_CAP_FILE_SAVEAS
+      - basic block:
+        - and:
+          - api: SendMessage
+          - number: 0x414 = WM_CAP_FILE_SET_CAPTURE_FILE
```
Comment (williballenthin) [inline nursery/capture-webcam-video.yml:25]:
```suggestion
    - and:
      - os: windows
      - api: capCreateCaptureWindow
      - basic block:
        - and:
          - api: SendMessage
          - number: 0x43E = WM_CAP_SEQUENCE
      - or:          
        - basic block:
          - and:
            - api: SendMessage
            - number: 0x417 = WM_CAP_FILE_SAVEAS
        - basic block:
          - and:
            - api: SendMessage
            - number: 0x414 = WM_CAP_FILE_SET_CAPTURE_FILE
```

> ... it seems like we could look for calling capCreateCaptureWindow + calling capCaptureSequence + (calling capFileSaveAs or calling capFileSetCaptureFile)

Comment (williballenthin) [reply nursery/capture-webcam-video.yml:25]: two changes:
  - restrict to `os: windows`
  - use logic from original issue, unless you have a better idea?

File: nursery/capture-webcam-video.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    att&ck:
+      - Collection::Video Capture [T1125]
+  features:
+    - or:
+      - and:
+        - api: capCreateCaptureWindow
+        - basic block:
+          - and:
+            - api: SendMessage
+            - number: 0x43E = WM_CAP_SEQUENCE
```
Comment (williballenthin) [inline nursery/capture-webcam-video.yml:17]: thanks for noticing that `capCaptureSequence` is a macro and gets compiled to `SendMessage/WM_CAP_SEQUENCE`

**Review by williballenthin — CHANGES_REQUESTED** [review]
recommend using logic from original issue or explaining why you changed it. otherwise, looks great!

File: nursery/capture-webcam-video.yml (hunk 1 above)
Comment (johnk3r) [reply nursery/capture-webcam-video.yml:25]: 1st I had forgotten about the "SO" filter. Thanks for reinforcing.
2nd Your logic got better. Thanks for the sugestion.

File: nursery/capture-webcam-video.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      - Collection::Video Capture [T1125]
+  features:
+    - and:
+      - os: windows
+      - api: capCreateCaptureWindow
+      - basic block:
+        - and:
+          - api: SendMessage
+          - number: 0x43E = WM_CAP_SEQUENCE
+      - or:
```
Comment (williballenthin) [inline nursery/capture-webcam-video.yml:18]:
```suggestion
      - or:
```

cleanup whitespace to make the linter happy and then i'll merge!

**Comment by williballenthin** [comment]
thank you!

**Pull request merged**